File: src/_1ProgrammingBasicsWithJavaMay2023/_3ConditionalStatementsAdvanced/_3MoreExercise/_04CarToGo.java
package _1ProgrammingBasicsWithJavaMay2023._3ConditionalStatementsAdvanced._3MoreExercise;

import java.util.Scanner;

public class _04CarToGo {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double budget = Double.parseDouble(scanner.nextLine());
        String season = scanner.nextLine();
        String classOfCar = "";
        String typeOfCar = "";
        double priceOfCar = 0.00;
        if (budget <= 100) {
            classOfCar = "Economy class";
            if (season.equals("Summer")) {
                typeOfCar = "Cabrio";
                priceOfCar = budget * 0.35;
            } else if (season.equals("Winter")) {
                typeOfCar = "Jeep";
                priceOfCar = budget * 0.65;
            }
        } else if (budget > 100 && budget <= 500) {
            classOfCar = "Compact class";
            if (season.equals("Summer")) {
                typeOfCar = "Cabrio";
                priceOfCar = budget * 0.45;
            } else if (season.equals("Winter")) {
                typeOfCar = "Jeep";
                priceOfCar = budget * 0.80;
            }
        } else if (budget > 500) {
            classOfCar = "Luxury class";
            typeOfCar = "Jeep";
            priceOfCar = budget * 0.90;
        }
        System.out.printf("%s%n%s - %.2f", classOfCar, typeOfCar, priceOfCar);
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_3ConditionalStatementsAdvanced/_3MoreExercise/_04CarToGo.js
function carToGo(budget, season) {
    const CABRIO_ECONOMY_PRICE_PERCENT = 0.35;
    const CABRIO_COMPACT_PRICE_PERCENT = 0.45;
    const JEEP_ECONOMY_PRICE_PERCENT = 0.65;
    const JEEP_COMPACT_PRICE_PERCENT = 0.80;
    const JEEP_LUXORY_PRICE_PERCENT = 0.90;

    let carClass;
    let carType;
    let carPrice = 0.00;

    if (season === "Summer") {
        carType = "Cabrio";
        if (budget <= 100) {
            carClass = "Economy class";
            carPrice = budget * CABRIO_ECONOMY_PRICE_PERCENT;
        } else if (budget <= 500) {
            carClass = "Compact class";
            carPrice = budget * CABRIO_COMPACT_PRICE_PERCENT;
        } else if (budget > 500) {
            carClass = "Luxury class";
            carType = "Jeep";
            carPrice = budget * JEEP_LUXORY_PRICE_PERCENT;
        }
    } else if (season === "Winter") {
        carType = "Jeep";
        if (budget <= 100) {
            carClass = "Economy class";
            carPrice = budget * JEEP_ECONOMY_PRICE_PERCENT;
        } else if (budget <= 500) {
            carClass = "Compact class";
            carPrice = budget * JEEP_COMPACT_PRICE_PERCENT;
        } else if (budget > 500) {
            carClass = "Luxury class";
            carPrice = budget * JEEP_LUXORY_PRICE_PERCENT;
        }
    }

    console.log(`${carClass}`);
    console.log(`${carType} - ${carPrice.toFixed(2)}`);
}